### PR TITLE
rsz: use only data buffers

### PR DESCRIPTION
**Pull request by AcKoucher**
For #5577.

### CODE REVIEW

File: src/rsz/src/Resizer.cc [hunk 1]
```diff
@@ -497,18 +498,31 @@ void Resizer::balanceRowUsage()
...
 
-void Resizer::findBuffers()
+void Resizer::findDataBuffers()
 {
   if (buffer_cells_.empty()) {
     LibertyLibraryIterator* lib_iter = network_->libertyLibraryIterator();
+    sta::PatternMatch clkbuf_pattern(".*CLKBUF.*",
+                                     /* is_regexp */ true,
+                                     /* nocase */ true,
+                                     /* Tcl_interp* */ nullptr);
```
Comment (QuantamHD) [inline src/rsz/src/Resizer.cc:508]: It feels like we should factor this out into a separate function. There's lots and lots of different names for these clock buffers.

In general though it feels like we should be able to identify them based on the difference between their rise and fall times. We should have one class of buffers where the delta is much bigger and one class where it's nearly zero. If we run a k-means of 2 I suspect we'll get what we want.

Comment (AcKoucher) [reply src/rsz/src/Resizer.cc:508]: I used as reference the strategy used in TritonCTS when we select the buffers that will be used for the clock tree.
See `TritonCTS::inferBufferList`. I opened this as a draft to propose a discussion as I don't feel that this change is close to an actual solid solution. @maliberty @precisionmoon Do you have any input?

Comment (precisionmoon) [reply src/rsz/src/Resizer.cc:508]: @QuantamHD, @AcKoucher, Yes it makes sense to look for buffers with similar rise/fall delays in the absence of special buffer names or attributes.   We want to do this for both CTS and RSZ engines.  For now, I suggest that we proceed with cell names and attributes and enhance it for rise/fall delays in the future.

Comment (maliberty) [reply src/rsz/src/Resizer.cc:508]: I looked and all the public PDKs would work with this (not all have clock buffers).  There are private PDKs that don't follow this pattern.  I agree it is ok for a first step.

Comment (maliberty) [reply src/rsz/src/Resizer.cc:508]: We could centralize the decision about what is a clock buffer in dbSta so it can be shared with CTS.

**Comment by maliberty** [comment]
The code looks fine but this will need a secure CI.

**Comment by AcKoucher** [comment]
Running secure-CI. I'm studying the failures in the regression tests.

**Comment by titan73** [comment]
What is purpose of the secure CI?

**Comment by maliberty** [comment]
> What is purpose of the secure CI?

To test on proprietary PDKs that PII has access to under NDA.

**Comment by maliberty** [comment]
Precision Innovation Inc.

File: src/rsz/src/Resizer.cc (hunk 1 above)
Comment (gadfort) [reply src/rsz/src/Resizer.cc:508]: It seems like proceeding with a half solution like the clockbuf pattern thing would be a poor idea in my mind. If the better solution is the look at rise and fall times, etc then this should wait until a centralize solution is available

**Pull request closed without merging**